operator/controllers/offering_controller.go

```diff
@@ -37,15 +41,67 @@ type OfferingReconciler struct {
 // +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=offerings,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=servicebroker.metabroker.suse.com,resources=offerings/status,verbs=get;update;patch
 
+const offeringReconcileTimeout = time.Second * 10
+
+// Reconcile reconciles an Offering resource.
 func (r *OfferingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("offering", req.NamespacedName)
+	ctx, cancel := context.WithTimeout(context.Background(), offeringReconcileTimeout)
+	defer cancel()
+
+	log := r.Log.WithValues("offering", req.NamespacedName)
+
+	offering := &servicebrokerv1alpha1.Offering{}
+	if err := r.Get(ctx, req.NamespacedName, offering); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Offering resource deleted")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	offeringNeedsUpdate := false
 
-	// your logic here
+	if len(offering.OwnerReferences) == 0 {
+		if err := r.setOwnership(ctx, &req, offering); err != nil {
```

Comment:
This calls [`SetControllerReference()`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.4/pkg/controller/controllerutil#SetControllerReference); that seems a bit off, as (if I'm reading the docs correctly) you can have non-controller owner references (because there can be only one controller reference, which is a special case of an owner reference, of which there may be many).  That means if we already have multiple non-controller owner references there would be an issue here.  Of course, I'm also unsure why this would be a controller reference here in the first place… I guess it makes sense that the offering will go away if the provider it needs goes away?

Do we need to requeue if we can't find the matching provider? After all, maybe it'll show up in a second…

Reply:
You described exactly how `SetControllerReference` works and, yes, this is the desired behaviour. I understand that `offering` *could* have another owner reference in terms of _Kubernetes_ design, but it doesn't in this case. At the same time, I can probably do better with the logic to check if it is already owned by the controller object.

As for the `requeue`, there's a problem in my logic... if it encounters an error, I should return it, which causes a `requeue` automatically. Thanks for catching that.

Reply:
I'll keep the current logic for now as we are not setting any other ownership at this point, so the way I'm currently doing is way simpler and cheaper as we don't need to fetch the owner via an API call. Any fancier verification can be done in the future if this changes.

For the error logic, I pushed https://github.com/SUSE/metabroker/pull/6/commits/4e6705963a81d8eb540f78e58e3398d141918ee9.